### PR TITLE
Fix the deployment failure with default `values.yaml`.

### LINKED ISSUE #13558: Unable to install via helm with default values.yaml.
### Description

Che fails to install using helm with a template error. This seems to be because after: https://github.com/eclipse/che/pull/13455/files
there aren't default values for `che.workspace.pluginRegistryUrl` and `che.workspace. devfileRegistryUrl ` set.

### Reproduction Steps

Che fails to install into kubernetes using helm:
```
helm upgrade --install che --namespace che --set cheImage=eclipse/che-server:nightly --set global.cheWorkspacesNamespace="che" --set global.ingressDomain=${CHE_DOMAIN}.nip.io ./
```
I'm basing my command from the installation instructions for Docker for desktop here:
https://che.eclipse.org/running-eclipse-che-on-kubernetes-using-docker-desktop-for-mac-5d972ed511e1

**OS and version:**    
Docker is:
Version 2.0.0.3 (31259)
Helm Version:
Client: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}

**Diagnostics:** 

The error is:
```
Release "che" does not exist. Installing it now.
Error: render error in "che/templates/configmap.yaml": template: che/templates/configmap.yaml:87:14: executing "che/templates/configmap.yaml" at <.Values.che.workspac...>: can't evaluate field workspace in type interface {}
```

This seems to be caused by the new values:
`che.workspace.pluginRegistryUrl` and `che.workspace. devfileRegistryUrl `
here:
https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/templates/configmap.yaml#L87

There are no values for these by default but if I uncomment:
https://github.com/eclipse/che/blob/master/deploy/kubernetes/helm/che/values.yaml#L70
then I am able to install successfully.

### PR DESCRIPTION
**Pull request by monaka**
This will fix eclipse/che#13558.

Signed-off-by: Masaki Muranaka <monaka@monami-ya.com>

### What does this PR do?

- Checks if `nil` to places referencing `che.*`.
- Quotes some URL values
 
### What issues does this PR fix or reference?

Issue
- fixes #13558 

PR
- caused by #13455 
- closes #13560 (alternative)

### CODE REVIEW

**Comment by monaka** [comment]
I checked this by calling `helm lint` and `helm install --dry-run --debug .` with both commented (default) and uncommented `vaule.yaml`

**Comment by musienko-maxim** [comment]
Can one of the admins verify this PR?

**Comment by sleshchenko** [comment]
ci-build

**Comment by monaka** [comment]
It looks the test failed at unrelated place...

**Comment by skabashnyuk** [comment]
ci-build

**Comment by ssh24** [comment]
Nice! Thanks for creating the alternative PR @monaka

**Comment by monaka** [comment]
Could anybody approve this?

**Comment by benoitf** [comment]
@skabashnyuk @l0rd @sleshchenko could you please review ?

File: deploy/kubernetes/helm/che/values.yaml [hunk 1]
```diff
@@ -67,7 +67,7 @@ global:
     fsGroup: 1724
   postgresDebugLogs: false
 
-#che:
+che: {}
```
Comment (sleshchenko) [inline deploy/kubernetes/helm/che/values.yaml:70]: @monaka Can not we do the same for workspace section?
If `workspace.devfileRegistryUrl` field is uncommented, should `{}` be removed?

Comment (monaka) [reply deploy/kubernetes/helm/che/values.yaml:70]: @sleshchenko Yes. `{}` must be removed then. This approach is often shown on charts in the official Helm repository. (Just one example, Traefik https://github.com/helm/charts/blob/4cbc02a65f7c1bfef4cd4b03e3556a289b2f2014/stable/traefik/values.yaml#L35-L39)

**Review by sleshchenko — APPROVED** [review]

**Comment by sleshchenko** [comment]
@monaka Thanks for this contribution :+1: 
Feel free to merge if you test it.

**Pull request merged**